PEP 7: Reword & split the "C dialect" section (#4557)
python/peps: peps/pep-0007.rst

   clean up someone else's mess (in true XP style).

C standards
===========

Follow the following standards.
For features that aren't in the relevant standard (like atomics or some C11

> features in public headers), use CPython-specific wrappers.

Reviewer: I had to follow the wikipedia link to optional features below to understand this. The point is that certain features are declared optional by the standard, and for those, if we want to use them, we must use CPython-specific headers. (But why not allow checking the standard's feature-check macros directly, if the main code just has to have two versions? Wrappers are useful if you'd be checking for the feature-check macros repeatedly (like with atomics) or if a shorthand macro could help not seeing the feature checks at all in the rest of the code.

When adding such wrappers, try to make them easy to adjust for unsupported
compilers.

* Python 3.11 and newer versions use C11 without `optional features

>   <https://en.wikipedia.org/wiki/C11_%28C_standard_revision%29#Optional_features>`_.

Reviewer 2: The advice is to use anonymous links unless the document reuses the target name:
```suggestion
  <https://en.wikipedia.org/wiki/C11_%28C_standard_revision%29#Optional_features>`__.
```

* The public C API should be compatible with C99 and C++.

>   (This is looser than what we guarantee users. Remember that this PEP allows
>   exceptions!)

Author: ```suggestion
  (This is more restrictive than what we guarantee users. Remember that this PEP
  allows exceptions!)
```

>   allows exceptions!)

Reviewer: I don't understand what this means. Maybe a brief example would clarify it?

Author: That's a note for users, who sometimes take this out of context. I'll clarify.

   clean up someone else's mess (in true XP style).

> C standards
> ===========

Reviewer 3: Retain the old anchor:

```suggestion
.. _c_dialect:

C standards
===========
```
